sunpycube/cube/NDCube.py

```diff
@@ -0,0 +1,127 @@
+            If a list it should contain one element for each axis of the numpy array.
+            For the image axes a [min, max] pair should be specified which will be
+            passed to :func:`matplotlib.pyplot.imshow` as extent.
+            For the slider axes a [min, max] pair can be specified or an array the
+            same length as the axis which will provide all values for that slider.
+            If None is specified for an axis then the array indices will be used
+            for that axis.
+        """
+        i = ImageAnimatorWCS(self.data, wcs=self.wcs, *args, **kwargs)
+        return i
+
+    def __getitem__(self, item):
+        if item is None or (isinstance(item, tuple) and None in item):
+            raise IndexError("None indices not supported")
+        data = self.data[item]
+        wcs = sunpycube.wcs_util._wcs_slicer(self.wcs, item)
+        if self.mask is not None:
+            mask = self.mask[item]
+        else:
+            mask = None
+        if data.ndim is 2:
+            result = Cube2D(data, wcs=wcs, mask=mask, uncertainty=self.uncertainty,
+                            meta=self.meta, unit=self.unit, copy=False)
+        elif data.ndim is 1:
+            result = Cube1D(data, wcs=wcs, mask=mask, uncertainty=self.uncertainty,
+                            meta=self.meta, unit=self.unit, copy=False)
+        else:
+            result = NDCube(data, wcs=wcs, mask=mask, uncertainty=self.uncertainty,
+                            meta=self.meta, unit=self.unit, copy=False)
+        return result
+
+
+class Cube2D(NDCube):
+    """docstring for Cube2D"""
+
+    def __init__(self, data, uncertainty=None, mask=None, wcs=None, meta=None, unit=None, copy=False, **kwargs):
+        super(Cube2D, self).__init__(data, uncertainty=uncertainty, mask=mask,
+                                     wcs=wcs, meta=meta, unit=unit, copy=copy, **kwargs)
+
+    def plot(self, axes=None, axis_data=['x', 'y'], **kwargs):
+        if axes is None:
+        	if self.wcs.naxis is not 2:
+        		slice_list = self.wcs._naxis
+        		axis_index = []
+        		for i, ax in enumerate(slice_list):
+        			if ax is not 1:
+        				axis_index.append(i)
+        		if len(axis_index) is 2:
+        			slice_list[axis_index[0]] = axis_data[0]
+        			slice_list[axis_index[1]] = axis_data[1]
+        		else:
+        			raise ValueError("Dimensions of WCS and data don't match")
+        	axes = wcsaxes_compat.gca_wcs(self.wcs, slices=slice_list)
+        plot = axes.imshow(self.data, **kwargs)
+        return plot
+
+
+class Cube1D(NDCube):
+    """docstring for Cube1D"""
+
+    def __init__(self, data, uncertainty=None, mask=None, wcs=None, meta=None, unit=None, copy=False, **kwargs):
+        super(Cube1D, self).__init__(data, uncertainty=uncertainty, mask=mask,
+                                     wcs=wcs, meta=meta, unit=unit, copy=copy, **kwargs)
+
+    def plot(self):
+        plot = plt.plot(self.data)
```

Comment:
You should be plotting this against physical coordinates. So you I'll want to do a call through `pixel_to_world` to get them. Also it's worth looking into the matplotlib support for astropy units.